notebook/services/kernels/handlers.py

```diff
@@ -244,6 +264,85 @@ def on_message(self, msg):
             return
         stream = self.channels[channel]
         self.session.send(stream, msg)
+        
+    def _on_zmq_reply(self, stream, msg_list):
+        
+        def write_stderr(error_message):
+            parent = json.loads(msg_list[-3])
+            msg = self.session.msg("stream",
+                content={"text": error_message, "name": "stderr"},
+                parent=parent
+            )
+            msg['channel'] = 'iopub'
+            self.write_message(json.dumps(msg, default=date_default))
+            self.log.warn(error_message)
+            
+        channel = getattr(stream, 'channel', None)
+        if channel == 'iopub':
+    
+            # If the elapsed time is double the check interval, reset the counts,
+            # lock flags, and reset the timer.
+            elapsed_time = (time.time() - self._last_limit_check)
+            if elapsed_time >= self.limit_poll_interval * 2.0:
+                self._iopub_msg_count = 0
+                self._iopub_byte_count = 0
+                if self._iopub_msgs_exceeded or self._iopub_data_exceeded:
+                    self._iopub_data_exceeded = False
+                    self._iopub_msgs_exceeded = False
+                    self.log.warn("iopub messages resumed")
+                
+            # Increment the bytes and message count
+            self._iopub_msg_count += 1
+            self._iopub_byte_count += sum([len(x) for x in msg_list])
+            
+            # If the elapsed time is equal to or greater than the limiter
+            # interval, check the limits, set the limit flags, and reset the
+            # message and data counts.
+            if elapsed_time >= self.limit_poll_interval:
```

Comment:
Sure, I think that is a good idea.  What I ended up doing was changing the rate poll interval to a rate checking window.  I no longer reset the message or byte counts, instead I keep two queues for changing the counts sometime in the future.  Whenever I change the counts, I also add the opposite to the queue timed for now + window.  See 69faf0f